### PR TITLE
[RFC] Silent output when read from stdin

### LINKED ISSUE #5921: Silent output when read from stdin
- `nvim --version`: 0.1.7
- Operating system/version: Arch
- Terminal name/version: unicode-rxvt v9.22
- `$TERM`: rxvt-unicode-256color

### Actual behaviour
Extra file detail on quit.
Example output:
```
Nvim: Reading from stdin...
"-stdin-" 1L, 1C"-stdin-" 1L, 1C
```

### Expected behaviour
Only output `Nvim: Reading from stdin...`

Vim output:
```
Vim: Reading from stdin...

```

### Steps to reproduce using `nvim -u NORC`

```
echo | /usr/bin/nvim --noplugin -c ':q!' -
```

### PR DESCRIPTION
**Pull request by DarkDeepBlue**
This part of the code needed an improvement anyway, because while reading from `stdin`, `msg_trunc_attr()` is called twice, since `readfile()` is called twice (with `read_stdin == true` and `read_buffer == true`). Hence the doubled output.

This PR removes the output for `stdin` entirely and leaves only the desired message after exiting `nvim`.

Resolves #5921.

### CODE REVIEW

File: src/nvim/fileio.c [hunk 1]
```diff
@@ -707,7 +707,7 @@ readfile (
      * always using the GUI.
      */
     if (read_stdin) {
-      mch_msg(_("Nvim: Reading from stdin...\n"));
+      mch_msg(_("Nvim: Reading from stdin ...\n"));
```
Comment (justinmk) [inline src/nvim/fileio.c]: The problem with changing translated messages (wrapped in `_("...")`) is that all the .po files must be updated too :(

**Comment by justinmk** [comment]
Merged. Thank you @DarkDeepBlue

File: src/nvim/fileio.c [hunk 2]
```diff
@@ -1883,16 +1883,20 @@ readfile (
       xfree(keep_msg);
       keep_msg = NULL;
       msg_scrolled_ign = TRUE;
-      p = msg_trunc_attr(IObuff, FALSE, 0);
+
+      if (!read_stdin && !read_buffer) {
+        p = msg_trunc_attr(IObuff, FALSE, 0);
```
Comment (oni-link) [inline src/nvim/fileio.c]: Previously `p` was always set here. Now `p` can (in the else-branch) point to something freed and `set_keep_msg(p,0)` may be called with this pointer value. Should `p` be set to `NULL` in the else-branch?

```
==5448==ERROR: AddressSanitizer: heap-use-after-free on address 0x631000050884 at pc 0x00000046c232 bp 0x7ffe3fdb65e0 sp 0x7ffe3fdb5d90
READ of size 1 at 0x631000050884 thread T0
    #0 0x46c231 in __interceptor_strlen.part.31 /home/oni-link/git/llvm/projects/compiler-rt/lib/asan/../sanitizer_common/sanitizer_common_interceptors.inc:284
    #1 0xf0c706 in xstrdup /home/oni-link/git/neovim/src/nvim/memory.c:401:24
    #2 0x16e1bbd in vim_strsave /home/oni-link/git/neovim/src/nvim/strings.c:51:20
    #3 0xf29e1e in set_keep_msg /home/oni-link/git/neovim/src/nvim/message.c:1017:16
    #4 0xc79d95 in readfile /home/oni-link/git/neovim/src/nvim/fileio.c:1898:9
    #5 0x655f3e in open_buffer /home/oni-link/git/neovim/src/nvim/buffer.c:176:16
    #6 0xdf01c1 in read_stdin /home/oni-link/git/neovim/src/nvim/main.c:1337:9
    #7 0xde1d6d in main /home/oni-link/git/neovim/src/nvim/main.c:418:5
    #8 0x7f765c839730 in __libc_start_main (/lib64/libc.so.6+0x20730)
    #9 0x4473b8 in _start (/home/oni-link/git/neovim/build/bin/nvim+0x4473b8)

0x631000050884 is located 132 bytes inside of 65537-byte region [0x631000050800,0x631000060801)
freed by thread T0 here:
    #0 0x4fcb20 in __interceptor_cfree.localalias.1 /home/oni-link/git/llvm/projects/compiler-rt/lib/asan/asan_malloc_linux.cc:54
    #1 0xf0b834 in xfree /home/oni-link/git/neovim/src/nvim/memory.c:130:3
    #2 0xc6ffe8 in readfile /home/oni-link/git/neovim/src/nvim/fileio.c:1012:9
    #3 0x655f3e in open_buffer /home/oni-link/git/neovim/src/nvim/buffer.c:176:16
    #4 0xdf01c1 in read_stdin /home/oni-link/git/neovim/src/nvim/main.c:1337:9
    #5 0xde1d6d in main /home/oni-link/git/neovim/src/nvim/main.c:418:5
    #6 0x7f765c839730 in __libc_start_main (/lib64/libc.so.6+0x20730)

previously allocated by thread T0 here:
    #0 0x4fccd8 in malloc /home/oni-link/git/llvm/projects/compiler-rt/lib/asan/asan_malloc_linux.cc:64
    #1 0xf0b594 in try_malloc /home/oni-link/git/neovim/src/nvim/memory.c:84:15
    #2 0xf0b5d4 in verbose_try_malloc /home/oni-link/git/neovim/src/nvim/memory.c:100:15
    #3 0xc6fe7f in readfile /home/oni-link/git/neovim/src/nvim/fileio.c:1001:24
    #4 0x655f3e in open_buffer /home/oni-link/git/neovim/src/nvim/buffer.c:176:16
    #5 0xdf01c1 in read_stdin /home/oni-link/git/neovim/src/nvim/main.c:1337:9
    #6 0xde1d6d in main /home/oni-link/git/neovim/src/nvim/main.c:418:5
    #7 0x7f765c839730 in __libc_start_main (/lib64/libc.so.6+0x20730)

SUMMARY: AddressSanitizer: heap-use-after-free /home/oni-link/git/llvm/projects/compiler-rt/lib/asan/../sanitizer_common/sanitizer_common_interceptors.inc:284 in __interceptor_strlen.part.31

```

Comment (DarkDeepBlue) [inline src/nvim/fileio.c]: Oh, yes! While writing this, I set `p = NULL` at the beginning of the function without checking where it is actually used and I thought that would do it. Lession learned.

I will write a small PR for this as soon as I can.

Comment (justinmk) [inline src/nvim/fileio.c]: @oni-link Thank you. @DarkDeepBlue see https://github.com/neovim/neovim/commit/c86caf7e41c77f85861fc92e1ef1462508d74b24

**Pull request closed without merging**